homeassistant/components/rhasspy/stt.py

```diff
@@ -0,0 +1,197 @@
+
+        try:
+            # First chunk is a WAV header (no frames)
+            header_chunk = True
+            with io.BytesIO() as wav_io:
+                wav_file = wave.open(wav_io, "wb")
+                async for audio_chunk, _ in stream.iter_chunks():
+                    if header_chunk:
+                        # Extract WAV information
+                        header_chunk = False
+                        with io.BytesIO(audio_chunk) as header_io:
+                            with wave.open(header_io) as header_file:
+                                wav_file.setnchannels(header_file.getnchannels())
+                                wav_file.setsampwidth(header_file.getsampwidth())
+                                wav_file.setframerate(header_file.getframerate())
+                    else:
+                        # Everything after first chunk is audio data.
+                        # Add to in-memory WAV file.
+                        wav_file.writeframes(audio_chunk)
+
+                wav_file.close()
+                wav_data = wav_io.getvalue()
+                _LOGGER.debug("Received %s byte(s)", len(wav_data))
+
+            # POST to Rhasspy server
+            session = async_get_clientsession(self.hass)
+            async with session.post(
+                self.speech_url, headers=self.headers, data=wav_data
+            ) as res:
+                res.raise_for_status()
+                text_result = await res.text()
+                _LOGGER.debug(text_result)
+
+                return SpeechResult(text=text_result, result=SpeechResultState.SUCCESS)
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("async_process_audio_stream")
+
+        return SpeechResult(text="", result=SpeechResultState.ERROR)
+
+    # -------------------------------------------------------------------------
+
+    @property
+    def supported_languages(self) -> List[str]:
+        """Return a list of supported languages."""
+        return SUPPORT_LANGUAGES
+
+    @property
+    def supported_formats(self) -> List[AudioFormats]:
+        """Return a list of supported formats."""
+        return [AudioFormats.WAV]
+
+    @property
+    def supported_codecs(self) -> List[AudioCodecs]:
+        """Return a list of supported codecs."""
+        return [AudioCodecs.PCM]
+
+    @property
+    def supported_bit_rates(self) -> List[AudioBitRates]:
+        """Return a list of supported bitrates."""
+        return list(AudioBitRates)
+
+    @property
+    def supported_sample_rates(self) -> List[AudioSampleRates]:
+        """Return a list of supported samplerates."""
+        return list(AudioSampleRates)
+
+    @property
+    def supported_channels(self) -> List[AudioChannels]:
+        """Return a list of supported channels."""
+        return list(AudioChannels)
+
+
+# -----------------------------------------------------------------------------
+
+
+class RhasspyWavView(http.HomeAssistantView):
```

Comment:
Please remove this, that is not an part of an STT platform.

Reply:
Removed